src/ArrayOfArraysView.hpp

```diff
@@ -796,9 +796,22 @@ class ArrayOfArraysView
         INDEX_TYPE const totalSize = m_offsets[ newSize ];
 
         INDEX_TYPE const maxOffset = m_offsets[ m_numArrays ];
-        typeManipulation::forEachArg( [totalSize, maxOffset]( auto & buffer )
+        typeManipulation::forEachArg( [this, totalSize]( auto & buffer )
         {
-          bufferManipulation::reserve( buffer, maxOffset, MemorySpace::host, totalSize );
+          // We create a new buffer to avoid moving from uninitialized values.
+          auto newBuffer = std::remove_reference_t< decltype( buffer ) >( true );
+          bufferManipulation::reserve( newBuffer, 0, MemorySpace::host, totalSize );
+
+          for( INDEX_TYPE array = 0; array < m_numArrays; ++array )
+          {
+            INDEX_TYPE const curArraySize = sizeOfArray( array );
+            INDEX_TYPE const curArrayOffset = m_offsets[ array ];
+            arrayManipulation::uninitializedMove( &newBuffer[ curArrayOffset ], curArraySize, &buffer[ curArrayOffset ] );
+            arrayManipulation::destroy( &buffer[ curArrayOffset ], curArraySize );
+          }
+          
+          buffer.free();
+          buffer = std::move( newBuffer );
```

Comment:
This is to prevent doing things to uninitialized values that we shouldn't be doing. Only really a problem for `ArrayOfArrays<std::string>`.